compiler/mir/proto_mir.nim

```diff
@@ -286,15 +291,107 @@ func isStable(e: seq[ProtoItem], n: int): bool =
   else:
     unreachable(e[n].kind)
 
+func ownershipOp(e: seq[ProtoItem], i: int): ProtoItemKind =
+  ## Infers and returns the best fitting operation to retrieve an owning
+  ## value from the given *lvalue*.
+  proc decayMove(kind: ProtoItemKind): ProtoItemKind {.inline.} =
+    # moving from a projection requires a destructive move, since the source
+    # location needs to be destroyed after (in order to free the non-moved
+    # parts)
+    case kind
+    of pirMove: pirDestructiveMove
+    else:       kind
+
+  case e[i].kind
+  of pirParam:
+    if e[i].typ.kind == tySink:
+      pirSink
+    else:
+      pirCopy
+  of pirLocal, pirGlobal:
+    if sfCursor in e[i].sym.flags:
```

Comment:
That's a very good question. While it might be a bit restrictive, I do think it'd be better to necessitate a copy, like you said, at least for now. Ultimately, it depends on what assumptions the compiler should be allowed to make about an `importc`'ed location.

When the expectations/requirements become clearer, the necessitated copy could be loosened, so that:
```nim
var external {.importc.}: seq[int]
external = @[1, 2, 3]
var other = external # this could move, since `external` is immediately reassigned after 
external = @[2, 3, 4]
var other2 = external # cannot move, since the compiler doesn't know how/where/if
                      # `external` is used afterwards
```

Since forcing a copy would be a change in semantics, I'd do this as a follow-up, if that's okay with you.

Reply:
Thanks for the detailed answer, the change in semantics should not be part of this PR, it's really about the refactor